### ISSUE TITLE
[docs] [API] TRUNCATE documentation doesn't mention seemingly supported CASCADE / RESTRICT clauses

### ISSUE THREAD

**Issue by lukaseder** [yugabyte/yugabyte-db#9949]
The documentation for the `TRUNCATE` statement doesn't document the seemingly supported `CASCADE` / `RESTRICT` clauses. Both of these work:

```sql
create table t (i int);
truncate table t cascade;
truncate table t restrict;
```

Since PostgreSQL compatibility is a primary goal of Yugabyte, I'd say this needs to be either documented, or rejected by the parser as not yet supported.

See: https://docs.yugabyte.com/latest/api/ysql/the-sql-language/statements/ddl_truncate/#syntax

**Comment by polarweasel** [comment]
@bllewell can you take a look at this one, please?

**Comment by bllewell** [comment]
Fixed by [PR #11386](https://github.com/yugabyte/yugabyte-db/pull/11386).
